pyartcd/pyartcd/pipelines/ocp4.py

```diff
@@ -809,18 +812,44 @@ async def run(self):
             self.runtime.logger.info('Building only where source has changed.')
             await self._plan_builds()
 
-        await self._rebase_and_build_rpms()
+        await self._run_with_distgit_rebase_lock(self._rebase_and_build_rpms())
+
         if not self.skip_plashets:
             await self._build_compose()
         else:
             self.runtime.logger.warning('Skipping plashets creation as SKIP_PLASHETS was set to True')
-        await self._rebase_images()
+
+        await self._run_with_distgit_rebase_lock(self._rebase_images())
+
         await self._build_images()
         await self._sync_images()
         await self._mirror_rpms()
         await self._sweep()
         self._report_success()
 
+    async def _run_with_distgit_rebase_lock(self, coro: coroutine):
+        """
+        Tries to acquire github-activity-lock-4.y, then executes func()
+        """
+
+        if self.ignore_locks:
+            return await coro
+
+        lock = Lock.DISTGIT_REBASE
+        lock_manager = locks.LockManager.from_lock(lock)
+        lock_name = lock.value.format(version=self.version.stream)
+
+        async with await lock_manager.lock(lock_name):
+            try:
+                return await coro
+
+            except LockError as e:
+                self.runtime.logger.error('Failed acquiring lock %s: %s', e)
+                raise
+
+            finally:
+                await lock_manager.destroy()
```

Comment:
Here, are we destroying the connection before the context manager releases the lock?

Reply:
Right, thanks for catching this. Fixed